optaplanner-core/src/main/java/org/optaplanner/core/api/score/stream/ConstraintCollectors.java

```diff
@@ -197,6 +199,29 @@
                 resultContainer -> resultContainer[0]);
     }
 
+    public static <A> UniConstraintCollector<A, ?, Duration> sum(Function<? super A, Duration> groupValueMapping) {
```

Comment:
By adding these methods, previously compiling code no longer does due to ambiguity.

Reply:
I really don't think that this is a way forward. Adding a new function `sumSomething()` for every `Something` somebody may want to sum, that simply does not scale. 

We can approach it from a different angle: 

`<X> sum(Function<A, X>, Function<X, X> adder)`

And if you want to be extra nice to people:

`apply(Sum.duration())`, where `Sum.duration()` is a convenience implementation we offer to people, and we could also provide averages etc. in a similar manner.

Reply:
In  Java Stream API (and nearly all like it), this operation is called `reduce()`:  (https://docs.oracle.com/javase/8/docs/api/java/util/stream/package-summary.html#Reduction). Although in streams, a map is done before the reduce (so it would look like `stream.map(toDuration).reduce(sumBiOperation)`)